portal/server/helpers/constructPayload.js

```diff
@@ -5,16 +5,24 @@
  * @param {Object} body Request body (req.body)
  * @param {Array} properties Interested properties to be added into payload
  * @param {Boolean} csrf Include CSRF token, defaulted to `true`
+ * @param {Array} unsetIfPropertyIsEmpty The condition to determine whether to delete the property
  * @returns {Object} Payload
  */
-const constructPayload = (body, properties, csrf = true) => {
+const constructPayload = (body, properties, unsetIfPropertyIsEmpty = [], csrf = true) => {
   let payload = {};
-
+  const newObj = { ...body };
   // Return empty payload upon void mandatory arguments
   if (!body || !properties) {
     return payload;
   }
 
+  // Currency will be an object, don't save as empty string, because Mongo DB can't change type.
+  for (const property of unsetIfPropertyIsEmpty) {
```

Comment:
Refactor this if condition please.

Reply:
We should prevent an empty property in a payload being added to the database and a validation error should be thrown on the same page preventing a flawed payload being added to the database.

Reply:
Error exists for BSS and EWCS facilities, for them at the moment we allow skipping some values in forms and coming back to them later. **Also some fields are optional**.

I agree that we could try avoid saving empty values to DB, not sure it will require a lot changes, maybe code is expecting empty values.
@abhi-markan throwing errors would be big change in BSS/EWCS behavior, do we want it? is it bug related?

Reply:
Refactored `if` condition